File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.py
# def solution(n):
#     if (n ** (1/2))%1 == 0:
#         return (n ** (1/2) + 1) ** 2
#     else:
#         return -1
    
import math
def solution(n):
    x = math.sqrt(n)
    # if (n ** (1/2))%1 == 0: #혹은 x == int(x) 를 사용해서 정수를 확인할 수 있습니다.
    if x == int(x):
        return math.pow(x+1, 2)
    return -1

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        double srt = Math.pow(n, 0.5);
        boolean isNumber = srt%1 == 0.0;
        
        return isNumber?(long) Math.pow(srt + 1, 2):-1;
    }
}
